Mathlib/MeasureTheory/Measure/SeparableMeasure.lean

```diff
@@ -0,0 +1,486 @@
+    intro s ms hμs ε ε_pos
+    have mono : Monotone (fun n ↦ (T n) ∩ s) := fun m n hmn ↦ inter_subset_inter_left s
+        (biUnion_subset_biUnion_left fun k hkm ↦ Nat.le_trans hkm hmn)
+    have := tendsto_measure_iUnion (μ := μ) mono
+    rw [← tendsto_toReal_iff] at this
+    · -- We can therefore choose `N` such that `μ s - μ ((S N) ∩ s) < ε/2`.
+      rcases Metric.tendsto_atTop.1 this (ε / 2) (by linarith [ε_pos]) with ⟨N, hN⟩
+      have : Fact (μ (T N) < ∞) := Fact.mk <| T_finite N
+      -- Then we can apply the previous result to the measure `μ ((S N) ∩ •)`.
+      -- There exists `t ∈ 𝒜` such that `μ ((S N) ∩ (s ∆ t)) < ε/2`.
+      rcases (measureDense_of_generateFrom_setAglebra_of_finite
+        (μ := μ.restrict (T N)) h𝒜 hgen).approx s ms
+        (ne_of_lt (lt_of_le_of_lt (μ.restrict_apply_le _ s) hμs.lt_top))
+        (ε / 2) (by linarith [ε_pos])
+        with ⟨t, t_mem, ht⟩
+      -- We can then use `t ∩ (S N)`, because `S N ∈ 𝒜` by hypothesis.
+      -- `μ (s ∆ (t ∩ S N))`
+      --   `≤ μ (s ∆ (s ∩ S N)) + μ ((s ∩ S N) ∆ (t ∩ S N))`
+      --   `= μ s - μ (s ∩ S N) + μ (s ∆ t) ∩ S N) < ε`.
+      refine ⟨t ∩ T N, h𝒜.inter_mem t_mem (T_mem N), ?_⟩
+      calc
+        μ (s ∆ (t ∩ T N))
+          ≤ μ (s \ (s ∩ T N)) + μ ((s ∆ t) ∩ T N) := by
+              rw [← symmDiff_of_le (inter_subset_left ..), symmDiff_comm _ s,
+                inter_symmDiff_distrib_right]
+              exact measure_symmDiff_le _ _ _
+        _ < ENNReal.ofReal (ε / 2) + ENNReal.ofReal (ε / 2) := by
+              apply ENNReal.add_lt_add
+              · rw [measure_diff
+                    (inter_subset_left ..)
+                    (ms.inter (hgen ▸ measurableSet_generateFrom (T_mem N)))
+                    (ne_top_of_le_ne_top hμs (measure_mono (inter_subset_left ..))),
+                  lt_ofReal_iff_toReal_lt (sub_ne_top hμs),
+                  toReal_sub_of_le (measure_mono (inter_subset_left ..)) hμs]
+                apply lt_of_le_of_lt (sub_le_dist ..)
+                nth_rw 1 [← univ_inter s]
+                rw [inter_comm s, dist_comm, ← T_spanning, iUnion_inter _ T]
+                apply hN N (le_refl _)
+              · rwa [← μ.restrict_apply' (hgen ▸ measurableSet_generateFrom (T_mem N))]
+        _ = ENNReal.ofReal ε := by
+              rw [← ofReal_add (by linarith [ε_pos]) (by linarith [ε_pos]), add_halves]
+    · exact fun n ↦ ne_top_of_le_ne_top hμs (measure_mono (inter_subset_right ..))
+    · exact ne_top_of_le_ne_top hμs
+        (measure_mono (iUnion_subset (fun i ↦ inter_subset_right ..)))
+
+end MeasureDense
+
+section IsSeparable
+
+/-! ### Definition of a separable measure space, sufficient condition -/
+
+/-- A measure `μ` is separable if there exists a countable and measure-dense family of sets.
+
+The term "separable" is justified by the fact that the definition translates to the usual notion
+of separability in the metric space made by constant indicators equipped with the `L¹` norm. -/
```

Comment:
You keep saying this, but I don't see it *explicitly* anywhere in this PR. However, (a) it's true not just for `L¹`, but for `Lᵖ` more generally, right? And (b), I think this is essentially proved in part of the `Lp.induction` step in the second countability instance. So, I think it makes sense to factor that out as a lemma.

Reply:
I added `Measure.MeasureDense.indicatorConstLp_subset_closure`, which indeed factors out a part of the computations, tell me if it satisfies you. I did not add an equivalent result for separability because it seemed less useful, but tell me if I should add it.